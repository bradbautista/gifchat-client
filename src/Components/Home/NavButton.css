.card-container {
    height: 26rem;
    margin-bottom: 5rem;
    position: relative;
}

.card {
    display: flex;
    flex-direction: column;
    justify-content: center;
    position: absolute;
    padding: 3rem;
    width: 100%;
    outline: 2px solid var(--secondary);
    z-index: 5;
    cursor: pointer;
}
  
.front {
    background-color: var(--quaternary);
    z-index: 0;
    bottom: 0px;
    top: 0px;
    left: 0px;
    right: 0px;
    font-size: 3rem;
    font-weight: 900;
    color: black;
}
  
.back {
    background-color: var(--quinary);
    z-index: -6;
    top: -10px;
    left: -10px;
    bottom: 10px;
    right: 10px;
    font-size: 2rem;
    transition: 0.5s;
    letter-spacing: -0.2px;
}
  
.swap {
    transition: all 0.5s;
    z-index: 6;
    top: 10px;
    left: 10px;
    right: -10px;
    bottom: -10px;
}

/* TABLET BREAKPOINT */
  
@media screen and (min-width: 720px) {

    .front {
        font-size: 4.5rem;
        padding-left: 7rem;
    }

    .back {
        font-size: 2.5rem;
    }
    
}

/* iPad pro BREAKPOINT */

@media screen and (min-width: 1024px) {

    .card-container {
        height: 30rem;
    }

    .front {
        font-size: 6rem;
        padding-left: 7rem;
    }

    .back {
        font-size: 3.5rem;
    }

}


/* DESKTOP BREAKPOINT */
  
@media screen and (min-width: 1200px) {

    .card-container {
        height: 17rem;
        margin-bottom: 5rem;
    }

    .front {
        font-size: 3rem;
    }

    .back {
        font-size: 2rem;
    }
    
}