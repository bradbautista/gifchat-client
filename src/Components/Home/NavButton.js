import React, { Component } from 'react'
import './NavButton.css'

export default class NavButton extends Component {

  constructor(props) {
    super(props);
    this.state = {
      swapped: false,
    };

  }

  toggleSwap = () => {
    const currentState = this.state.swapped;
    this.setState({ swapped: !currentState})
  }

  render() {

    return (        

        <div className="card-container">
            <div 
            onClick={() => {this.props.onClick(); this.toggleSwap()}}
            // onClick={(this.props.frontCardContent == 'Get a rando') ? this.props.onClick : this.toggleSwap} 
            className="card front"
            >
                {this.props.frontCardContent}
            </div>

            <div 
            // onClick={this.toggleSwap} 
            className={'card' + ' ' + 'back' + ' ' + ((this.state.swapped) ? 'swap' : null)}>
                {this.props.backCardContent}
            </div>
        </div>



            
    );
  }  
}
