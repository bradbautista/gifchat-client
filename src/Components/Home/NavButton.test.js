import React from 'react';
import ReactDOM from 'react-dom';
import NavButton from './NavButton';
import renderer from 'react-test-renderer';
import { Link } from 'react-router-dom';
import { BrowserRouter } from 'react-router-dom';

it('Renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<NavButton />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('Renders the "Get a room" button as expected', () => {
    const someRoom = 'a-big-red-dog-named-Clifford'
    const tree = renderer
        .create(
            <BrowserRouter>
            <NavButton 
                frontCardContent="Get a room" 
                backCardContent={
                    <>
                        <span>Room created at </span>
                        <Link
                            to={`/rooms/${someRoom}`}
                            className="room-link"
                            target="_blank"
                            rel="noopener noreferrer"
                        >
                            {someRoom}.
                        </Link>
                    </>
                }
            />
            </BrowserRouter>
            )
        .toJSON();
    expect(tree).toMatchSnapshot();
        
})

it('Renders the "Go to room" button as expected', () => {
    const someFunc = () => {};
    const someRoom = 'a-big-red-dog-named-Clifford';
    const tree = renderer
        .create(
            <BrowserRouter>
            <NavButton
              onClick={someFunc}
              frontCardContent="Go to room"
              backCardContent={
                <>
                  <span className="input-prompt">
                    Enter room name, i.e. a-big-red-dog-named-Clifford
                  </span>
                  <div className="home-input-flex-wrapper">
                    <input
                      className="destination-field"
                      value={someRoom}
                      onChange={someFunc}
                      type="text"
                      placeholder="a-big-red-dog-named-Clifford"
                    />
                    <button
                      className="go-button"
                      onClick={() => {
                        this.openInNewTab(`rooms/${someRoom}`);
                      }}
                    >
                      Go
                    </button>
                  </div>
                </>
              }
            />
            </BrowserRouter>
            )
        .toJSON();
    expect(tree).toMatchSnapshot();
        
})

it('Renders the "Get a rando" button as expected', () => {
    const someFunc = () => {};
    const tree = renderer
        .create(
            <BrowserRouter>
            <NavButton
              onClick={() => {
                someFunc;
              }}
              frontCardContent="Get a rando"
              backCardContent="Searching for randos. If none are found, you will be put in a room to wait until one comes along."
            />
            </BrowserRouter>
            )
        .toJSON();
    expect(tree).toMatchSnapshot();
        
})