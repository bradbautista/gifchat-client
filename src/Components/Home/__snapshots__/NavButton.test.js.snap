// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the "Get a rando" button as expected 1`] = `
<div
  className="card-container"
>
  <div
    className="card front"
    onClick={[Function]}
  >
    Get a rando
  </div>
  <div
    className="card back null"
  >
    Searching for randos. If none are found, you will be put in a room to wait until one comes along.
  </div>
</div>
`;

exports[`Renders the "Get a room" button as expected 1`] = `
<div
  className="card-container"
>
  <div
    className="card front"
    onClick={[Function]}
  >
    Get a room
  </div>
  <div
    className="card back null"
  >
    <span>
      Room created at 
    </span>
    <a
      className="room-link"
      href="/rooms/a-big-red-dog-named-Clifford"
      onClick={[Function]}
      rel="noopener noreferrer"
      target="_blank"
    >
      a-big-red-dog-named-Clifford
      .
    </a>
  </div>
</div>
`;

exports[`Renders the "Go to room" button as expected 1`] = `
<div
  className="card-container"
>
  <div
    className="card front"
    onClick={[Function]}
  >
    Go to room
  </div>
  <div
    className="card back null"
  >
    <span
      className="input-prompt"
    >
      Enter room name, i.e. a-big-red-dog-named-Clifford
    </span>
    <div
      className="home-input-flex-wrapper"
    >
      <input
        className="destination-field"
        onChange={[Function]}
        placeholder="a-big-red-dog-named-Clifford"
        type="text"
        value="a-big-red-dog-named-Clifford"
      />
      <button
        className="go-button"
        onClick={[Function]}
      >
        Go
      </button>
    </div>
  </div>
</div>
`;
